src/operator/nn/concat.cc

```diff
@@ -320,5 +375,41 @@ NNVM_REGISTER_OP(_backward_Concat)
 #endif
 .set_attr<FCompute>("FCompute<cpu>", ConcatGradCompute<cpu>);
 
+
+NNVM_REGISTER_OP(_rnn_param_concat)
```

Comment:
Please write comments for this operator. How is it different from the normal concat.

Reply:
This is a custom concat op with specialized infer_shape, which handles the case where the first one or two inputs may have unknown shape that can be inferred from output shape.